lua/focus/modules/split.lua

```diff
@@ -6,138 +6,140 @@ local M = {}
+            -- this is not useful here but keeping as this might be useful to halve window sizes in future
+            -- vim.api.nvim_win_set_height(winnr, vim.api.nvim_win_get_height(winnr)/2)
+        else
+            vim.o.minwinwidth = vim.o.minwinwidth / 2
+        end
+    end
+
+    -- splitright.. ensure that whenever you split vertically, it’s going to appear on the right.
+    -- Moreover, for a horizontal split, the new split is going to appear at the bottom.
+    -- always returns false/nil below when not set by user
+    -- we check it because when splitbelow/right is set, wincmd happens automatically
+    if split_cmd == 'vsplit' and not vim.o.splitright then
+        cmd('wincmd p')
+    end
+
+    if split_cmd == 'split' and not vim.o.splitbelow then
+        cmd('wincmd p')
+    end
+
+    --open file or term .. etc if args provided
+    process_split_args(true, args, bufnew)
 end
 
 function M.split_command(direction, args, tmux, bufnew)
-	local winnr = vim.api.nvim_get_current_win()
-
-	local created = false
-	if M.split_exists_direction(winnr, direction) == false then
-		created = true
-		if tmux == true then
-			vim.fn.system('tmux select-pane -' .. vim.fn.tr(direction, 'phjkl', 'lLDUR'))
-		elseif direction == 'h' or direction == 'l' then
-			cmd('wincmd v')
-			cmd('wincmd ' .. direction)
-		elseif direction == 'j' or direction == 'k' then
-			cmd('wincmd s')
-			cmd('wincmd ' .. direction)
-		end
-	else
-		cmd('wincmd ' .. direction)
-	end
-	process_split_args(created, args, bufnew)
+    local winnr = vim.api.nvim_get_current_win()
+
+    local created = false
+    if M.split_exists_direction(winnr, direction) == false then
+        created = true
+        if tmux == true then
+            vim.fn.system(
+                'tmux select-pane -' .. vim.fn.tr(direction, 'phjkl', 'lLDUR')
+            )
+        elseif direction == 'h' or direction == 'l' then
+            cmd('wincmd v')
+            cmd('wincmd ' .. direction)
+        elseif direction == 'j' or direction == 'k' then
+            cmd('wincmd s')
+            cmd('wincmd ' .. direction)
+        end
+    else
+        cmd('wincmd ' .. direction)
+    end
+    process_split_args(created, args, bufnew)
 end
 
 function M.split_cycle(reverse, bufnew)
-	local winnr = vim.api.nvim_get_current_win()
-	if reverse == 'reverse' then
-		cmd('wincmd W')
-	else
-		cmd('wincmd w')
-	end
-
-	if winnr == vim.api.nvim_get_current_win() then
-		cmd('wincmd v')
-		if reverse == nil or reverse ~= 'reverse' then
-			cmd('wincmd w')
-		end
-		if bufnew == true then
-			cmd('enew')
-		end
-	end
+    local winnr = vim.api.nvim_get_current_win()
+    if reverse == 'reverse' then
+        cmd('wincmd W')
+    else
+        cmd('wincmd w')
+    end
+
+    if winnr == vim.api.nvim_get_current_win() then
+        cmd('wincmd v')
+        if reverse == nil or reverse ~= 'reverse' then
+            cmd('wincmd w')
+        end
+        if bufnew == true then
+            cmd('enew')
```

Comment:
Should the created buffer be a scratch buffer so it disappears when the user opens a file in that split?